<a href="https://colab.research.google.com/github/SabrinaDantas/projeto_programacao_para_dados_pucrs/blob/main/projeto_programacaoparadados_fase02.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Bibliotecas

In [1]:
import pandas as pd

# Leitura do Arquivo

In [2]:
df_steam = pd.read_csv('/content/steam_games.csv')

In [3]:
df_steam.head(5)

AppID                   Name  Release date Estimated owners  Peak CCU  \
0    20200       Galactic Bowling  Oct 21, 2008        0 - 20000         0   
1   655370           Train Bandit  Oct 12, 2017        0 - 20000         0   
2  1732930           Jolt Project  Nov 17, 2021        0 - 20000         0   
3  1355720               Henosis™  Jul 23, 2020        0 - 20000         0   
4  1139950  Two Weeks in Painland   Feb 3, 2020        0 - 20000         0   

   Required age  Price  DLC count  \
0             0  19.99          0   
1             0   0.99          0   
2             0   4.99          0   
3             0   5.99          0   
4             0   0.00          0   

                                      About the game  \
0  Galactic Bowling is an exaggerated and stylize...   
1  THE LAW!! Looks to be a showdown atop a train....   
2  Jolt Project: The army now has a new robotics ...   
3  HENOSIS™ is a mysterious 2D Platform Puzzler w...   
4  ABOUT THE GAME Play as a hacker who has arrang...   

                                 Supported languages  ...  \
0                                        ['English']  ...   
1  ['English', 'French', 'Italian', 'German', 'Sp...  ...   
2                 ['English', 'Portuguese - Brazil']  ...   
3  ['English', 'French', 'Italian', 'German', 'Sp...  ...   
4                     ['English', 'Spanish - Spain']  ...   

  Average playtime two weeks Median playtime forever  \
0                          0                       0   
1                          0                       0   
2                          0                       0   
3                          0                       0   
4                          0                       0   

  Median playtime two weeks             Developers             Publishers  \
0                         0  Perpetual FX Creative  Perpetual FX Creative   
1                         0           Rusty Moyher           Wild Rooster   
2                         0          Campião Games          Campião Games   
3                         0      Odd Critter Games      Odd Critter Games   
4                         0          Unusual Games          Unusual Games   

                                          Categories  \
0  Single-player,Multi-player,Steam Achievements,...   
1  Single-player,Steam Achievements,Full controll...   
2                                      Single-player   
3              Single-player,Full controller support   
4                   Single-player,Steam Achievements   

                            Genres  \
0              Casual,Indie,Sports   
1                     Action,Indie   
2  Action,Adventure,Indie,Strategy   
3           Adventure,Casual,Indie   
4                  Adventure,Indie   

                                                Tags  \
0                        Indie,Casual,Sports,Bowling   
1  Indie,Action,Pixel Graphics,2D,Retro,Arcade,Sc...   
2                                                NaN   
3  2D Platformer,Atmospheric,Surreal,Mystery,Puzz...   
4  Indie,Adventure,Nudity,Violent,Sexual Content,...   

                                         Screenshots  \
0  https://cdn.akamai.steamstatic.com/steam/apps/...   
1  https://cdn.akamai.steamstatic.com/steam/apps/...   
2  https://cdn.akamai.steamstatic.com/steam/apps/...   
3  https://cdn.akamai.steamstatic.com/steam/apps/...   
4  https://cdn.akamai.steamstatic.com/steam/apps/...   

                                              Movies  
0  http://cdn.akamai.steamstatic.com/steam/apps/2...  
1  http://cdn.akamai.steamstatic.com/steam/apps/2...  
2  http://cdn.akamai.steamstatic.com/steam/apps/2...  
3  http://cdn.akamai.steamstatic.com/steam/apps/2...  
4  http://cdn.akamai.steamstatic.com/steam/apps/2...  

[5 rows x 39 columns]

In [4]:
df_steam.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 72934 entries, 0 to 72933
Data columns (total 39 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   AppID                       72934 non-null  int64  
 1   Name                        72932 non-null  object 
 2   Release date                72934 non-null  object 
 3   Estimated owners            72934 non-null  object 
 4   Peak CCU                    72934 non-null  int64  
 5   Required age                72934 non-null  int64  
 6   Price                       72934 non-null  float64
 7   DLC count                   72934 non-null  int64  
 8   About the game              70402 non-null  object 
 9   Supported languages         72934 non-null  object 
 10  Full audio languages        72934 non-null  object 
 11  Reviews                     9217 non-null   object 
 12  Header image                72934 non-null  object 
 13  Website                     354

# Pergunta 1

Quais são os dez jogos mais bem avaliados, de acordo com o Metacritic? No caso de notas repetidas, ordenar os jogos de acordo com suas datas de lançamento (do mais velho para o mais recente).

In [5]:
df_steam['Release date'] = pd.to_datetime(df_steam['Release date'])

In [6]:
df_steam.head(5)

AppID                   Name Release date Estimated owners  Peak CCU  \
0    20200       Galactic Bowling   2008-10-21        0 - 20000         0   
1   655370           Train Bandit   2017-10-12        0 - 20000         0   
2  1732930           Jolt Project   2021-11-17        0 - 20000         0   
3  1355720               Henosis™   2020-07-23        0 - 20000         0   
4  1139950  Two Weeks in Painland   2020-02-03        0 - 20000         0   

   Required age  Price  DLC count  \
0             0  19.99          0   
1             0   0.99          0   
2             0   4.99          0   
3             0   5.99          0   
4             0   0.00          0   

                                      About the game  \
0  Galactic Bowling is an exaggerated and stylize...   
1  THE LAW!! Looks to be a showdown atop a train....   
2  Jolt Project: The army now has a new robotics ...   
3  HENOSIS™ is a mysterious 2D Platform Puzzler w...   
4  ABOUT THE GAME Play as a hacker who has arrang...   

                                 Supported languages  ...  \
0                                        ['English']  ...   
1  ['English', 'French', 'Italian', 'German', 'Sp...  ...   
2                 ['English', 'Portuguese - Brazil']  ...   
3  ['English', 'French', 'Italian', 'German', 'Sp...  ...   
4                     ['English', 'Spanish - Spain']  ...   

  Average playtime two weeks Median playtime forever  \
0                          0                       0   
1                          0                       0   
2                          0                       0   
3                          0                       0   
4                          0                       0   

  Median playtime two weeks             Developers             Publishers  \
0                         0  Perpetual FX Creative  Perpetual FX Creative   
1                         0           Rusty Moyher           Wild Rooster   
2                         0          Campião Games          Campião Games   
3                         0      Odd Critter Games      Odd Critter Games   
4                         0          Unusual Games          Unusual Games   

                                          Categories  \
0  Single-player,Multi-player,Steam Achievements,...   
1  Single-player,Steam Achievements,Full controll...   
2                                      Single-player   
3              Single-player,Full controller support   
4                   Single-player,Steam Achievements   

                            Genres  \
0              Casual,Indie,Sports   
1                     Action,Indie   
2  Action,Adventure,Indie,Strategy   
3           Adventure,Casual,Indie   
4                  Adventure,Indie   

                                                Tags  \
0                        Indie,Casual,Sports,Bowling   
1  Indie,Action,Pixel Graphics,2D,Retro,Arcade,Sc...   
2                                                NaN   
3  2D Platformer,Atmospheric,Surreal,Mystery,Puzz...   
4  Indie,Adventure,Nudity,Violent,Sexual Content,...   

                                         Screenshots  \
0  https://cdn.akamai.steamstatic.com/steam/apps/...   
1  https://cdn.akamai.steamstatic.com/steam/apps/...   
2  https://cdn.akamai.steamstatic.com/steam/apps/...   
3  https://cdn.akamai.steamstatic.com/steam/apps/...   
4  https://cdn.akamai.steamstatic.com/steam/apps/...   

                                              Movies  
0  http://cdn.akamai.steamstatic.com/steam/apps/2...  
1  http://cdn.akamai.steamstatic.com/steam/apps/2...  
2  http://cdn.akamai.steamstatic.com/steam/apps/2...  
3  http://cdn.akamai.steamstatic.com/steam/apps/2...  
4  http://cdn.akamai.steamstatic.com/steam/apps/2...  

[5 rows x 39 columns]

In [7]:
top_10 = df_steam.sort_values(by=['Metacritic score', 'Release date'], ascending=[False, True])

In [8]:
top_10[['AppID', 'Name', 'Release date', 'Metacritic score']].head(10)

AppID                           Name Release date  Metacritic score
45752   632470  Disco Elysium - The Final Cut   2019-10-15                97
64935  1687950                Persona 5 Royal   2022-10-20                97
49557       70                      Half-Life   1998-11-08                96
36775      220                    Half-Life 2   2004-11-16                96
32406     7670                      BioShock™   2007-08-21                96
57205   271590             Grand Theft Auto V   2015-04-13                96
28170      620                       Portal 2   2011-04-18                95
28484      659                       Portal 2   2011-04-18                95
12746     3900   Sid Meier's Civilization® IV   2006-10-25                94
55150    34440   Sid Meier's Civilization® IV   2006-10-25                94

# Pergunta 2

Para jogos de role-playing, qual o número médio e máximo de: DLCs, avaliações positivas, avaliações negativas e materiais de demonstração (número de capturas de tela e filmes, somados)?  



In [9]:
def contar_itens(row):
  if pd.isna(row['Screenshots']) and pd.isna(row['Movies']):
    return 0
  elif pd.isna(row['Screenshots']) or pd.isna(row['Movies']):
    return 1
  else:
    return 2

In [10]:
df_steam['Demo materials'] = df_steam.apply(contar_itens, axis=1)


In [11]:
df_steam[['Screenshots', 'Movies', 'Demo materials']].sample(25)

Screenshots  \
59184  https://cdn.akamai.steamstatic.com/steam/apps/...   
67429  https://cdn.akamai.steamstatic.com/steam/apps/...   
4022   https://cdn.akamai.steamstatic.com/steam/apps/...   
56811  https://cdn.akamai.steamstatic.com/steam/apps/...   
15297  https://cdn.akamai.steamstatic.com/steam/apps/...   
39349  https://cdn.akamai.steamstatic.com/steam/apps/...   
37815  https://cdn.akamai.steamstatic.com/steam/apps/...   
55879  https://cdn.akamai.steamstatic.com/steam/apps/...   
4354   https://cdn.akamai.steamstatic.com/steam/apps/...   
69141                                                NaN   
27128  https://cdn.akamai.steamstatic.com/steam/apps/...   
44049  https://cdn.akamai.steamstatic.com/steam/apps/...   
21160  https://cdn.akamai.steamstatic.com/steam/apps/...   
67848  https://cdn.akamai.steamstatic.com/steam/apps/...   
35647                                                NaN   
58432  https://cdn.akamai.steamstatic.com/steam/apps/...   
4071   https://cdn.akamai.steamstatic.com/steam/apps/...   
51232  https://cdn.akamai.steamstatic.com/steam/apps/...   
65633  https://cdn.akamai.steamstatic.com/steam/apps/...   
53344  https://cdn.akamai.steamstatic.com/steam/apps/...   
29987  https://cdn.akamai.steamstatic.com/steam/apps/...   
30463  https://cdn.akamai.steamstatic.com/steam/apps/...   
34774  https://cdn.akamai.steamstatic.com/steam/apps/...   
11238  https://cdn.akamai.steamstatic.com/steam/apps/...   
29286  https://cdn.akamai.steamstatic.com/steam/apps/...   

                                                  Movies  Demo materials  
59184  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
67429  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
4022   http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
56811  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
15297  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
39349  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
37815  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
55879  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
4354   http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
69141                                                NaN               0  
27128  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
44049  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
21160  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
67848  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
35647                                                NaN               0  
58432  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
4071   http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
51232  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
65633  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
53344  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
29987  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
30463  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
34774  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
11238  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
29286  http://cdn.akamai.steamstatic.com/steam/apps/2...               2

In [12]:
df_steam['Genres'].sample(15)

17972                                          RPG
39559         Adventure,Casual,Simulation,Strategy
19758                                       Action
26344                                   Simulation
3514                                Violent,Action
66431              Design & Illustration,Utilities
37595                              Casual,Strategy
46806                              Adventure,Indie
36844                             Casual,Indie,RPG
17355                 Adventure,Indie,Early Access
6046                             Casual,Simulation
37587         Casual,Free to Play,Indie,Simulation
34886                                    Adventure
18858    Strategy,Education,Utilities,Early Access
62598                        Casual,Indie,Strategy
Name: Genres, dtype: object

In [13]:
jogos_rpg = df_steam[df_steam['Genres'].str.contains('RPG', na=False)]
jogos_rpg['Genres'].sample(15)

25057                    Action,Adventure,Casual,Indie,RPG
60272    Action,Adventure,Free to Play,Indie,RPG,Early ...
49192                                            Indie,RPG
71198                                                  RPG
16400              Action,Adventure,Indie,RPG,Early Access
32690                                 Action,Adventure,RPG
4320                                                   RPG
72790                           Action,Adventure,Indie,RPG
18573                                        Adventure,RPG
35268    Action,Indie,Massively Multiplayer,RPG,Early A...
27478         Action,Adventure,Casual,Indie,RPG,Simulation
32173    Action,Adventure,Casual,Indie,RPG,Simulation,S...
54246                                 Action,Adventure,RPG
55049                    Action,Adventure,Casual,Indie,RPG
6833           Adventure,Indie,RPG,Simulation,Early Access
Name: Genres, dtype: object

In [14]:
# Número médio e máximo de DLCs
media_dlcs = jogos_rpg['DLC count'].mean()
maximo_dlcs = jogos_rpg['DLC count'].max()

print(f"Número médio de DLCs do gênero RPG: {media_dlcs:.2f}")
print(f"Número máximo de DLCs do gênero RPG: {maximo_dlcs}")
print('\n')

# Número médio e máximo de avaliações positivas
media_positivas = jogos_rpg['Positive'].mean()
maximo_positivas = jogos_rpg['Positive'].max()

print(f"Número médio de avaliações positivas do gênero RPG: {media_positivas:.2f}")
print(f"Número máximo de avaliações positivas do gênero RPG: {maximo_positivas}")
print('\n')

# Número médio e máximo de avaliações negativas
media_negativas = jogos_rpg['Negative'].mean()
maximo_negativas = jogos_rpg['Negative'].max()

print(f"Número médio de avaliações negativas do gênero RPG: {media_negativas:.2f}")
print(f"Número máximo de avaliações negativas do gênero RPG: {maximo_negativas}")
print('\n')

# Número médio e máximo de materiais de demonstração
media_demo = jogos_rpg['Demo materials'].mean()
maximo_demo = jogos_rpg['Demo materials'].max()

print(f"Número médio de materiais de demonstração do gênero RPG: {media_demo:.2f}")
print(f"Número máximo de materiais de demonstração do gênero RPG: {maximo_demo}")

Número médio de DLCs do gênero RPG: 0.95
Número máximo de DLCs do gênero RPG: 2366


Número médio de avaliações positivas do gênero RPG: 1516.41
Número máximo de avaliações positivas do gênero RPG: 964983


Número médio de avaliações negativas do gênero RPG: 247.17
Número máximo de avaliações negativas do gênero RPG: 129925


Número médio de materiais de demonstração do gênero RPG: 1.95
Número máximo de materiais de demonstração do gênero RPG: 2


# Pergunta 3

Quais são as cinco empresas que mais publicam jogos pagos na plataforma? Para tais empresas, qual o número médio e mediano de avaliações positivas de seus jogos pagos?

In [15]:
# Jogos Pagos
jogos_pagos = df_steam[df_steam['Price'] > 0]

In [16]:
# Agrupar por 'Publishers' e calcular o número total de avaliações positivas
avaliacoes_por_editor = jogos_pagos.groupby('Publishers')['Positive'].sum()

In [17]:
#Cinco empresas com mais jogos pagos
cinco_maiores_editores = avaliacoes_por_editor.nlargest(5)
print("As cinco empresas que mais publicam jogos pagos na plataforma são", ", ".join(cinco_maiores_editores.index.to_list()))

As cinco empresas que mais publicam jogos pagos na plataforma são Ubisoft, Valve, Bethesda Softworks, Electronic Arts, Xbox Game Studios


In [18]:
# Número médio e mediano de avaliações positivas para essas empresas
cinco_empresas = ["Ubisoft", "Valve", "Bethesda Softworks", "Electronic Arts", "Xbox Game Studios"]

# Loop sobre cada empresa
for empresa in cinco_empresas:
    # Filtrar o DataFrame original para incluir apenas jogos da empresa atual
    jogos_empresa = jogos_pagos[jogos_pagos['Publishers'] == empresa]

    # Calcular média e mediana de avaliações positivas para a empresa atual
    media_avaliacoes = jogos_empresa['Positive'].mean()
    mediana_avaliacoes = jogos_empresa['Positive'].median()

    # Imprimir estatísticas para a empresa atual
    print(f"Estatísticas para {empresa}:")
    print(f"Número médio de avaliações positivas: {media_avaliacoes:.2f}")
    print(f"Número mediano de avaliações positivas: {mediana_avaliacoes:.0f}")
    print()

Estatísticas para Ubisoft:
Número médio de avaliações positivas: 26678.76
Número mediano de avaliações positivas: 1712

Estatísticas para Valve:
Número médio de avaliações positivas: 110762.15
Número mediano de avaliações positivas: 24758

Estatísticas para Bethesda Softworks:
Número médio de avaliações positivas: 37584.28
Número mediano de avaliações positivas: 11147

Estatísticas para Electronic Arts:
Número médio de avaliações positivas: 14694.99
Número mediano de avaliações positivas: 4795

Estatísticas para Xbox Game Studios:
Número médio de avaliações positivas: 27110.40
Número mediano de avaliações positivas: 7872



# Pergunta 4

O número de jogos que suportam o sistema operacional Linux cresceu entre 2018 e 2022?

In [19]:
# jogos que suportam Linux
jogos_linux = df_steam[df_steam['Linux'] == True].copy()

In [20]:
jogos_linux['Year'] = jogos_linux['Release date'].dt.year
jogos_linux.head(1)

AppID      Name Release date Estimated owners  Peak CCU  Required age  \
3  1355720  Henosis™   2020-07-23        0 - 20000         0             0   

   Price  DLC count                                     About the game  \
3   5.99          0  HENOSIS™ is a mysterious 2D Platform Puzzler w...   

                                 Supported languages  ...  \
3  ['English', 'French', 'Italian', 'German', 'Sp...  ...   

  Median playtime two weeks         Developers         Publishers  \
3                         0  Odd Critter Games  Odd Critter Games   

                              Categories                  Genres  \
3  Single-player,Full controller support  Adventure,Casual,Indie   

                                                Tags  \
3  2D Platformer,Atmospheric,Surreal,Mystery,Puzz...   

                                         Screenshots  \
3  https://cdn.akamai.steamstatic.com/steam/apps/...   

                                              Movies  Demo materials  Year  
3  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  2020  

[1 rows x 41 columns]

In [21]:
# Filtrar os jogos lançados entre 2018 e 2022
jogos_2018_2022_linux = jogos_linux[jogos_linux['Year'].between(2018, 2022)]

In [22]:
# Contar o número de jogos lançados em cada ano entre 2018 e 2022
num_jogos_por_ano = jogos_2018_2022_linux['Year'].value_counts()
num_jogos_por_ano

2022    1311
2021    1211
2018    1190
2020    1084
2019     924
Name: Year, dtype: int64

In [23]:
# ano e o número de jogos lançados em ordem descrente dos anos
lista_num_jogos_ano = [(ano, num_jogos_por_ano[ano]) for ano in sorted(num_jogos_por_ano.index)]
lista_num_jogos_ano

[(2018, 1190), (2019, 924), (2020, 1084), (2021, 1211), (2022, 1311)]

Em 2018, foram lançados 1190 jogos que suportam o Linux. Em 2019, o número de jogos que suportam o Linux diminuiu para 924. No entanto, a partir de 2020, observamos um aumento consistente no número de jogos que suportam o Linux: 1084 em 2020, 1211 em 2021 e 1311 em 2022. Com base nesses números, podemos concluir que os jogos que suportam o sistema operacional Linux cresceu significativamente ao longo do período de 2018 a 2022

# Pergunta 5

Quais são os três jogos com as maiores recomendações que oferecem suporte ao idioma Português - BR e têm classificação etária superior a 16 anos?

In [24]:
# Suporta o idioma PT-BR
jogos_pt_br = df_steam[df_steam['Supported languages'].str.contains('Portuguese - Brazil', na=False)]
jogos_pt_br.sample(2)

AppID                Name Release date Estimated owners  Peak CCU  \
55479  1804450  Kingdom of Maverta   2021-11-11        0 - 20000         3   
224    1308880          FORECLOSED   2021-08-12        0 - 20000         0   

       Required age  Price  DLC count  \
55479             0   1.99          0   
224               0  16.99          0   

                                          About the game  \
55479  Kingdom of Maverta - minimalist, relaxing, mot...   
224    FORECLOSED is a narrative-driven action-advent...   

                                     Supported languages  ...  \
55479  ['English', 'French', 'Italian', 'German', 'Sp...  ...   
224    ['English', 'French', 'German', 'Spanish - Spa...  ...   

      Median playtime forever Median playtime two weeks            Developers  \
55479                       0                         0  My Label Game Studio   
224                         0                         0          Antab Studio   

                 Publishers  \
55479  My Label Game Studio   
224             Merge Games   

                                              Categories  \
55479                   Single-player,Steam Achievements   
224    Single-player,Steam Achievements,Full controll...   

                       Genres  \
55479                   Indie   
224    Action,Adventure,Indie   

                                                    Tags  \
55479  Casual,Puzzle,Side Scroller,Isometric,Atmosphe...   
224    Futuristic,Atmospheric,Sci-fi,Story Rich,Singl...   

                                             Screenshots  \
55479  https://cdn.akamai.steamstatic.com/steam/apps/...   
224    https://cdn.akamai.steamstatic.com/steam/apps/...   

                                                  Movies  Demo materials  
55479  http://cdn.akamai.steamstatic.com/steam/apps/2...               2  
224    http://cdn.akamai.steamstatic.com/steam/apps/2...               2  

[2 rows x 40 columns]

In [25]:
jogos_pt_br_16 = jogos_pt_br[jogos_pt_br['Required age'] >= 16]


In [26]:
jogos_pt_br_16 = jogos_pt_br_16.sort_values(by='Recommendations', ascending=False)


jogos_pt_br_16[['Name', 'Required age', 'Supported languages', 'Recommendations']].head(3)

Name  Required age  \
57205               Grand Theft Auto V            17   
38967  Tom Clancy's Rainbow Six® Siege            17   
21190  Tom Clancy's Rainbow Six® Siege            17   

                                     Supported languages  Recommendations  
57205  ['English', 'French', 'Italian', 'German', 'Sp...          1247051  
38967  ['English', 'French', 'Italian', 'German', 'Sp...           899838  
21190  ['English', 'French', 'Italian', 'German', 'Sp...           899613